File: example/src/main/java/com/factor/example/Adapter.java
package com.factor.example;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.factor.bouncy.BouncyRecyclerView;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;

public class Adapter extends BouncyRecyclerView.Adapter
{
    final ArrayList<String> data = new ArrayList<>();


    public Adapter()
    {
        for (int i = 0; i < 20; i ++)
            data.add("data: " + i);
    }

    @Override
    public void onItemMoved(int fromPosition, int toPosition) {

    }

    @Override
    public void onItemSwipedToStart(@Nullable RecyclerView.ViewHolder viewHolder, int positionOfItem) {

    }

    @Override
    public void onItemSwipedToEnd(@Nullable RecyclerView.ViewHolder viewHolder, int positionOfItem) {

    }

    @Override
    public void onItemSelected(@Nullable RecyclerView.ViewHolder viewHolder) {

    }

    @Override
    public void onItemReleased(@Nullable RecyclerView.ViewHolder viewHolder) {

    }

    @NonNull
    @NotNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {

        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item, parent, false);
        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull RecyclerView.ViewHolder holder, int position)
    {

    }

    @Override
    public int getItemCount()
    {
        return data.size();
    }


    static class MyViewHolder extends RecyclerView.ViewHolder
    {
        public MyViewHolder(@NonNull @NotNull View itemView)
        {
            super(itemView);
        }
    }
}

File: example/src/main/java/com/factor/example/Adapter.kt
package com.factor.example

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.factor.bouncy.BouncyRecyclerView
import java.util.*


class Adapter(size : Int) : BouncyRecyclerView.Adapter<Adapter.MyViewHolder>()
{

    private val data = ArrayList<String>()

    init
    {
        for (i in 1..size) data.add("data: $i")
    }


    override fun onItemMoved(fromPosition: Int, toPosition: Int)
    {
        Collections.swap(data, fromPosition, toPosition)
        notifyItemMoved(fromPosition, toPosition)
    }

    override fun onItemSwipedToStart(viewHolder: ViewHolder?, positionOfItem: Int)
    {
        data[positionOfItem] += "\nswiped to left"
        notifyItemChanged(positionOfItem)
    }

    override fun onItemSwipedToEnd(viewHolder: ViewHolder?, positionOfItem: Int)
    {
        data[positionOfItem] += "\nswiped to right"
        notifyItemChanged(positionOfItem)
    }

    override fun onItemSelected(viewHolder: ViewHolder?)
    {

    }

    override fun onItemReleased(viewHolder: ViewHolder?)
    {

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder
    {
        return MyViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item, parent, false))
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int)
    {
        holder.bindData(data[position])
    }

    override fun getItemCount(): Int
    {
        return data.size
    }


    class MyViewHolder(itemView: View) : ViewHolder(itemView)
    {
        fun bindData(data : String)
        {
            itemView.findViewById<TextView>(R.id.data_text).text = data
        }
    }


}
